frontend/src/pages/projects/notebook/NotebookStateStatus.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { Button, Label, LabelProps, Popover, Tooltip } from '@patternfly/react-core';
+import {
+  ExclamationCircleIcon,
+  InProgressIcon,
+  PauseCircleIcon,
+  RunningIcon,
+  SyncAltIcon,
+} from '@patternfly/react-icons';
+import { EventStatus } from '~/types';
+import { useDeepCompareMemoize } from '~/utilities/useDeepCompareMemoize';
+import { NotebookState } from './types';
+import { getEventFullMessage, useNotebookStatus } from './utils';
+import StartNotebookModal from './StartNotebookModal';
+
+type NotebookStateStatusProps = {
+  notebookState: NotebookState;
+  stopNotebook: () => void;
+};
+
+const rotate = 'pf-v5-c-spinner-animation-rotate 3s linear infinite';
+
+const NotebookStateStatus: React.FC<NotebookStateStatusProps> = ({
+  notebookState,
+  stopNotebook,
+}) => {
+  const { notebook, runningPodUid, isStarting, isStopping, isRunning } = notebookState;
+  const [unstableNotebookStatus, events] = useNotebookStatus(notebook, runningPodUid, isStarting);
+  const notebookStatus = useDeepCompareMemoize(unstableNotebookStatus);
+  const isError = notebookStatus?.currentStatus === EventStatus.ERROR;
+  const [isPopoverVisible, setPopoverVisible] = React.useState(false);
+  const [isStartModalOpen, setStartModalOpen] = React.useState(false);
+
+  const statusLabelSettings = React.useMemo((): {
+    label: string;
+    color: LabelProps['color'];
+    icon: React.ReactNode;
+  } => {
+    if (isError) {
+      return { label: 'Failed', color: 'red', icon: <ExclamationCircleIcon /> };
+    }
+    if (isStarting) {
+      return {
+        label: 'Starting',
+        color: 'blue',
+        icon: <InProgressIcon style={{ animation: rotate }} />,
+      };
+    }
+    if (isStopping) {
+      return {
+        label: 'Stopping',
+        color: 'grey',
+        icon: <SyncAltIcon style={{ animation: rotate }} />,
```

Comment:
We use these icons in several places.

PF doesn't have a means to spin icons? If not then, make then reusable components with their custom animation so we can update those other locations at a later date.